File: source_funcs/go_transcoder/WAYS_REMOVE_ONE_ELEMENT_BINARY_STRING_XOR_BECOMES_ZERO/WAYS_REMOVE_ONE_ELEMENT_BINARY_STRING_XOR_BECOMES_ZERO.go
package main

func f_gold(str []byte) int {
	var (
		one_count  int = 0
		zero_count int = 0
		n          int = int(len(str))
	)
	for i := int(0); i < n; i++ {
		if str[i] == byte('1') {
			one_count++
		} else {
			zero_count++
		}
	}
	if one_count%2 == 0 {
		return zero_count
	}
	return one_count
}

File: rust_funcs/go_transcoder/WAYS_REMOVE_ONE_ELEMENT_BINARY_STRING_XOR_BECOMES_ZERO/WAYS_REMOVE_ONE_ELEMENT_BINARY_STRING_XOR_BECOMES_ZERO.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::str;

fn f_gold(str: &str) -> i32 {
    let mut one_count = 0;
    let mut zero_count = 0;
    let n = str.len();
    for i in 0..n {
        if str.chars().nth(i).unwrap() == '1' {
            one_count += 1;
        } else {
            zero_count += 1; 
        }
    }
    if one_count % 2 == 0 {
        return zero_count; 
    }
    return one_count;
}
